pkg/router/template/router.go

```diff
@@ -203,6 +203,19 @@ func matchPattern(pattern, s string) bool {
 	return false
 }
 
+// Generate a regular expression to match wildcard hosts (and paths if any)
+// for a [sub]domain.
+func genDomainWildcardRegexp(hostname, path string, exactPath bool) string {
+	route := &routeapi.Route{Spec: routeapi.RouteSpec{Host: hostname}}
+	host, _ := routeapi.NormalizeWildcardHost(route.Spec.Host)
```

Comment:
Its a helper function for the template. Yeah its called from the template which calls it for wildcard routes.